### ISSUE TITLE
Weird issue with skip(?)

### ISSUE THREAD

**Issue by Vanawy** [eritislami/evobot#35]
1. I added [playlist](https://www.youtube.com/playlist?list=PLDb_G8HX8POTJMl1bmdmOfI9VHYNuRwtN)
2. Then skip first song
3. I get this in error.log
```
ReferenceError: Cannot access 'collector' before initialization
    at StreamDispatcher.<anonymous> (C:\Users\vanawy\Desktop\vanawy-evobot\include\play.js:34:9)
    at StreamDispatcher.emit (events.js:333:22)
    at finishMaybe (_stream_writable.js:698:14)
    at _stream_writable.js:674:7
    at StreamDispatcher._final (C:\Users\vanawy\Desktop\vanawy-evobot\node_modules\discord.js\src\client\voice\dispatcher\StreamDispatcher.js:248:5)
    at callFinal (_stream_writable.js:667:10)
    at processTicksAndRejections (internal/process/task_queues.js:84:21)
```

**Comment by Vanawy** [comment]
I think about adding additional check here
```
    const dispatcher = queue.connection
      .play(stream, { type: "opus" })
      .on("finish", () => {
        if (!collector.ended) collector.stop();
```

like this 

```
    const dispatcher = queue.connection
      .play(stream, { type: "opus" })
      .on("finish", () => {
        if (collector && !collector.ended) collector.stop();
```

but i dont test this yet

**Comment by eritislami** [comment]
I believe it should be fixed in 36c2a65 it only occurred when skip was used before the reactions promises were finished

**Comment by Vanawy** [comment]
Tested it locally and all works fine. Good job!
